### ISSUE TITLE
Some rules not running with Cypress

### ISSUE THREAD

**Issue by alisd23** [dequelabs/axe-core#1618]
**Expectation**: Run the `document-title` and `html-has-lang` rules when running axe-core inside Cypress (Which runs the whole app inside an iFrame)

**Actual**: The rules are identified as *inapplicable*

**Motivation**:
The `document-title` and `html-has-lang` rules should be able to run correctly when the whole app being tested is mounted inside an iFrame, which is what happens in the case of a Cypress test.

Currently I *think* what's happening is that these 2 rules use the `window-is-top.js` check, which causes the rule to only run for the **root/top** window object - i.e. not the actual window to be tested in a Cypress test.

Could there be a way to define a "root" or "top" window/document, so that this `window-is-top` code checks against a specific window object, instead of the default top window?

```
axe-core version: 3.2.2
Browser: Chrome
```

**Comment by straker** [comment]
Thanks for the issue. We'll have to investigate to see what can be done.

**Comment by dylanb** [comment]
@straker perhaps we could add an option that forces the running of the pageLevel rules, overriding the automatic detection

**Comment by straker** [comment]
So I've done some research. I verified that `window-is-top` is preventing the rules from running inside an iframe, which is what it [was suppose to do](https://github.com/dequelabs/axe-core/issues/1424). However, this prevents it from ever running inside an iframe even if that is what is desired (as is the case with Cypress). 

I think an option would work well. We can either provide an option to that tells axe it's being run inside an iframe and set it when axe injects itself into iframes, or an option that tells axe it's the top frame even if it's really not. The latter is more backwards compatible.

```js
axe.run({ isTopFrame: true });
```

**Comment by alisd23** [comment]
Sounds like a good solution to me. 👍 

Could you please give some quick advice/tips on where to start in adding this feature? If I have time in the future, and if no one picks it up before me - I might try give this a go.

**Comment by straker** [comment]
We'll need to get the passed in option to the `window-is-top` matcher so we can check for it as well. We currently don't pass any options into matcher functions, so you'd need to do this.

First, we automate the building of our function headers using a [template file](https://github.com/dequelabs/axe-core/blob/develop/build/templates.js#L5). Once you've modified the template to include options as a parameter, you'll need to [pass the options in the code](https://github.com/dequelabs/axe-core/blob/develop/lib/core/base/rule.js#L394). Lastly, [update the tests](https://github.com/dequelabs/axe-core/blob/develop/test/core/base/rule.js#L183-L233) to check that we pass the options parameter to the matches function.

**Comment by straker** [comment]
We might be able to just have `axe.run()` assume it's always running in the top-frame as a default, as well as an option to disable that assumption. In both cases, we should add the option to `context` so it's available within the matcher function `window-is-top`.

**Comment by padmavemulapati** [comment]
With Cypress - axe.run()
`injectAxe();
  injectIframes();
  cy.window({ log: false }).then(async (win: Window) => {
    const { includedImpacts, ...axeOptions } = options || {};
    win.axe.configure({ branding: { application: 'cypress' } });
    const results: axe.AxeResults = await win.axe.run(
      context || win.document,
      axeOptions
    );` we were able to run the above mentioned rules successfully getting the results properly.
